File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/JSCInstance.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.runtime;

import com.facebook.infer.annotation.Nullsafe;
import com.facebook.jni.HybridData;
import com.facebook.jni.annotations.DoNotStrip;
import com.facebook.soloader.SoLoader;

@Nullsafe(Nullsafe.Mode.LOCAL)
public class JSCInstance extends JSRuntimeFactory {
  static {
    SoLoader.loadLibrary("jscinstance");
  }

  @DoNotStrip
  protected static native HybridData initHybrid();

  public JSCInstance() {
    super(initHybrid());
  }
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/JSCInstance.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.runtime

import com.facebook.jni.HybridData
import com.facebook.jni.annotations.DoNotStrip
import com.facebook.jni.annotations.DoNotStripAny
import com.facebook.soloader.SoLoader

@DoNotStripAny
public class JSCInstance : JSRuntimeFactory(initHybrid()) {
  private companion object {
    init {
      SoLoader.loadLibrary("jscinstance")
    }

    @DoNotStrip @JvmStatic private external fun initHybrid(): HybridData
  }
}
